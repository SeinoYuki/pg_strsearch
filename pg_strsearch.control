# pg_strsearch extension
comment = 'pg_strsearch is add and submodule'
default_version = '1.0'
module_pathname = '$libdir/pg_strsearch'
relocatable = false
schema = pg_catalog
