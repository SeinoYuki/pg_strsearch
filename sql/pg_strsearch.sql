CREATE EXTENSION pg_strsearch;

CREATE TABLE pg_strsearch_rg (
	col1	VARCHAR(128),
	col2	VARCHAR(128),
	col3	VARCHAR(128)
);

INSERT INTO pg_strsearch_rg VALUES('test1', NULL, 'dummy');
INSERT INTO pg_strsearch_rg VALUES('test1', 'test2', 'dummy');
INSERT INTO pg_strsearch_rg VALUES('test1', 'test2', 'test3');

SELECT pg_strsearch('test');

DROP TABLE pg_strsearch_rg;
DROP EXTENSION pg_strsearch;
