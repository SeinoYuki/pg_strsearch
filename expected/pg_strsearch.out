CREATE EXTENSION pg_strsearch;
CREATE TABLE pg_strsearch_rg (
	col1	VARCHAR(128),
	col2	VARCHAR(128),
	col3	VARCHAR(128)
);
INSERT INTO pg_strsearch_rg VALUES('test1', NULL, 'dummy');
INSERT INTO pg_strsearch_rg VALUES('test1', 'test2', 'dummy');
INSERT INTO pg_strsearch_rg VALUES('test1', 'test2', 'test3');
SELECT pg_strsearch('test');
         pg_strsearch         
------------------------------
 (pg_strsearch_rg,col1,test1)
 (pg_strsearch_rg,col1,test1)
 (pg_strsearch_rg,col1,test1)
 (pg_strsearch_rg,col2,test2)
 (pg_strsearch_rg,col2,test2)
 (pg_strsearch_rg,col3,test3)
(6 rows)

DROP TABLE pg_strsearch_rg;
DROP EXTENSION pg_strsearch;
